File: test/Data.Analyzers.Vsix.Test/VisualBasic/ModelDesignerSpecRequiresArgument.vb
Imports DevZest.Data.Annotations

Public Class ModelDesignerSpecRequiresArgument
    Inherits Model

    Shared Sub New()
        RegisterColumn(Function(x As ModelDesignerSpecRequiresArgument) x.ID)
    End Sub

    Private m_ID As _Int32
    <DbColumn>
    Public Property ID As _Int32
        Get
            Return m_ID
        End Get
        Private Set
            m_ID = Value
        End Set
    End Property

End Class

File: test/Data.Analyzers.Vsix.Test/CSharp/ModelDesignerSpecRequiresArgument.cs
using DevZest.Data.Annotations;

namespace DevZest.Data.Analyzers.Vsix.Test.CSharp
{
    public class ModelDesignerSpecRequiresArgument : Model
    {
        static ModelDesignerSpecRequiresArgument()
        {
            RegisterColumn((ModelDesignerSpecRequiresArgument _) => _.Id);
        }

        [DbColumn]
        public _Int32 Id { get; private set; }
    }
}
